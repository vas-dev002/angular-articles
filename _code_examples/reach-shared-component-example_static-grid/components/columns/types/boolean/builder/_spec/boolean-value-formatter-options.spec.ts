import { ValueFormatterParams } from "@ag-grid-community/core";
import { TestBed } from "@angular/core/testing";
import { TranslateModule, TranslateService } from "@ngx-translate/core";
import { BooleanValueFormatterOptions } from "..";

declare const expect: jest.Expect;

describe("[Builder option]: BooleanValueFormatterOptions", () => {
	let booleanValueFormatterOptions: BooleanValueFormatterOptions;
	let translateService: TranslateService;

	beforeEach(() => {
		TestBed.configureTestingModule({
			imports: [TranslateModule.forRoot()],
			providers: [TranslateService]
		});

		translateService = TestBed.inject(TranslateService);
		booleanValueFormatterOptions = new BooleanValueFormatterOptions(translateService);
	});

	describe("[BooleanValueFormatterOptions instantiate]", () => {
		it("should be successfully created", () => {
			expect(booleanValueFormatterOptions).toBeTruthy();
			expect(booleanValueFormatterOptions).toMatchSnapshot();
		});
	});

	describe("[Method]: formatterFunction", () => {
		it("should return the currect value", () => {
			const params = { value: "True" };
			const expectedResult = params.value;
			const formatterFunction = booleanValueFormatterOptions.formatterFunction(params as unknown as ValueFormatterParams);
			expect(formatterFunction).toEqual(expectedResult);
		});

		it("should return null", () => {
			const params = { value: null };
			const formatterFunction = booleanValueFormatterOptions.formatterFunction(params as unknown as ValueFormatterParams);
			expect(formatterFunction).toBeNull();
		});
	});
});
