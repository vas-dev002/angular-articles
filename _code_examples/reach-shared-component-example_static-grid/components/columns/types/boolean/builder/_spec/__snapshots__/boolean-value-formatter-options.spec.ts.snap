// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Builder option]: BooleanValueFormatterOptions [BooleanValueFormatterOptions instantiate] should be successfully created 1`] = `
BooleanValueFormatterOptions {
  "formatterFunction": [Function],
  "langService": TranslateService {
    "_langs": Array [],
    "_onDefaultLangChange": EventEmitter_ {
      "__isAsync": false,
      "_isScalar": false,
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
    "_onLangChange": EventEmitter_ {
      "__isAsync": false,
      "_isScalar": false,
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
    "_onTranslationChange": EventEmitter_ {
      "__isAsync": false,
      "_isScalar": false,
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
    "_translationRequests": Object {},
    "_translations": Object {},
    "compiler": TranslateFakeCompiler {},
    "currentLoader": TranslateFakeLoader {},
    "extend": false,
    "isolate": false,
    "missingTranslationHandler": FakeMissingTranslationHandler {},
    "parser": TranslateDefaultParser {
      "templateMatcher": /\\{\\{\\\\s\\?\\(\\[\\^\\{\\}\\\\s\\]\\*\\)\\\\s\\?\\}\\}/g,
    },
    "pending": false,
    "store": TranslateStore {
      "currentLang": undefined,
      "langs": Array [],
      "onDefaultLangChange": EventEmitter_ {
        "__isAsync": false,
        "_isScalar": false,
        "closed": false,
        "hasError": false,
        "isStopped": false,
        "observers": Array [],
        "thrownError": null,
      },
      "onLangChange": EventEmitter_ {
        "__isAsync": false,
        "_isScalar": false,
        "closed": false,
        "hasError": false,
        "isStopped": false,
        "observers": Array [],
        "thrownError": null,
      },
      "onTranslationChange": EventEmitter_ {
        "__isAsync": false,
        "_isScalar": false,
        "closed": false,
        "hasError": false,
        "isStopped": false,
        "observers": Array [],
        "thrownError": null,
      },
      "translations": Object {},
    },
    "useDefaultLang": true,
  },
}
`;
